### PR TITLE
add missing vertical red line on reset

### LINKED ISSUE #107: No vertical red line appearing when resetting model during abmexploration
Here's a MWE pulled almost entirely from the AgentsZoo example page. 

According to this page https://juliadynamics.github.io/InteractiveDynamics.jl/dev/agents/#InteractiveDynamics.abmexploration
"Clicking the "reset" button will add a red vertical line to the data plots for visual guidance." But this doesn't happen for me.

I'm running InteractiveDynamics v0.21.11, Agents v5.4.3.

```julia
using InteractiveDynamics, Agents
using GLMakie
daisypath = joinpath(dirname(pathof(InteractiveDynamics)), "agents", "daisyworld_def.jl")
include(daisypath)
model, daisy_step!, daisyworld_step! = daisyworld(;
    solar_luminosity = 1.0, solar_change = 0.0, scenario = :change
)

daisycolor(a::Daisy) = a.breed # color of agents
as = 20 # size of agents
am = '✿' # marker of agents

params = Dict(
    :surface_albedo => 0:0.01:1,
    :solar_change => -0.1:0.01:0.1,
)

heatarray = :temperature
heatkwargs = (colorrange = (-20, 60), colormap = :thermal)
plotkwargs = (;
    ac = daisycolor, as, am,
    scatterkwargs = (strokewidth = 1.0,),
    heatarray, heatkwargs
)

using Statistics: mean
black(a) = a.breed == :black
white(a) = a.breed == :white
adata = [(black, count), (white, count)]
temperature(model) = mean(model.temperature)
mdata = [temperature, :solar_luminosity]
fig, abmobs = abmexploration(model;
    agent_step! = daisy_step!, model_step! = daisyworld_step!, params, plotkwargs...,
    adata, alabels = ["Black daisys", "White daisys"], mdata, mlabels = ["T", "L"]
)

display(fig)

if !isinteractive()
    println("Keeping process alive so that the Makie window doesn't close... Press ENTER to kill...")
    readline()
end
```

### PR DESCRIPTION
**Pull request by hbsmith**
fixes #107

### CODE REVIEW

File: src/agents/convenience.jl [hunk 1]
```diff
@@ -98,6 +98,11 @@ function init_abm_data_plots!(fig, p, adata, mdata, alabels, mlabels, plotkwargs
             ylims!(ax, Makie.MakieLayout.yautolimits(ax))
         end
     end
+    on(resetclick) do clicks
+        for ax in axs
+            vlines!(ax, [p.s.val], color = "#c41818")
```
Comment (fbanning) [inline src/agents/convenience.jl:103]: > #c41818

Was this the colour we've used in the earlier version?

**Review by fbanning — COMMENTED** [review]
Hi @hbsmith,

thanks for taking on this PR.  The changes look good to me. Happy to see that you were able to navigate the codebase of our interactive app recipe. :)

I suppose you already tested it on your machine and everything worked as expected. Would it be possible to post a quick screenshot of the result (just pick one of the example models from the docs for it)? That would be a nice way to document the changes.

File: src/agents/convenience.jl (hunk 1 above)
Comment (hbsmith) [reply src/agents/convenience.jl:103]: Yeah--I just found the last version of the code that had the colored line, and basically tried to modify it as little as possible to make it work after your refactor.

**Comment by hbsmith** [comment]
Sure, below are pictures before and after the code change.

Before: Screen Shot 2022-10-14 at 14 30 00

After: 
Screen Shot 2022-10-14 at 14 41 27

**Comment by Datseris** [comment]
Thanks, we appreciate this! And thanks @fbanning for the review!

**Pull request merged**